app/controllers/deploys_controller.rb

```diff
@@ -1,18 +1,13 @@
 class DeploysController < ApplicationController
-  include CurrentProject
   include ProjectLevelAuthorization
 
-
-  before_action except: [:active, :active_count, :recent, :changeset] do
-    find_project(params[:project_id])
-  end
-
```

Comment:
If you look in the routes.rb file, you'll find a lot of controllers are defined inside of the `resources :project` block.  That means that those endpoints should only ever get called if a valid project is specified.

So I propose:

1. In the `CurrentProject` module, have it add a `before_filter :find_project`.  That method throws an error if it can't find a valid project.
2. For any actions in controllers where we don't want to get the project, we add a `skip_before_filter` instruction.
3. The `find_project` method assumes `params[:project_id]`, since that's what is used in all but one controller.  The `ProjectsController` can override the method.

That way if there's an invalid project, that is detected immediately, rather than at some random point wherever `current_project` is first called.

That being said, I do like the idea of using `current_project` in the code rather than the `@project` instance variable.

What do you think of that idea?